In [15]:
from pandas import DataFrame
import pandas as pd

In [2]:
data= pd.read_csv("order_products__train.csv")

In [30]:
data.head()

order_id  product_id  add_to_cart_order  reordered
0         1       49302                  1          1
1         1       11109                  2          1
2         1       10246                  3          0
3         1       49683                  4          0
4         1       43633                  5          1

# Q1) Whats the product ID that's added to cart the most? 

In [34]:
data.sort_values(by='add_to_cart_order', ascending=False, na_position='first')

order_id  product_id  add_to_cart_order  reordered
1138788   2813632       41771                 80          1
567831    1395075       21641                 80          0
1138787   2813632       14862                 79          0
567830    1395075       47766                 79          1
567829    1395075       45063                 78          1
1138786   2813632        6187                 78          0
1138785   2813632       38400                 77          1
388834     949182       27086                 77          1
567828    1395075        1709                 77          0
139744     341238          79                 76          0
1138784   2813632       30806                 76          0
388833     949182       27845                 76          1
1161024   2869702       12440                 76          1
567827    1395075       44497                 76          1
567826    1395075       21868                 75          1
1138783   2813632       44325                 75          0
139743     341238       31506                 75          0
127783     312611       42148                 75          1
1161023   2869702       49628                 75          0
388832     949182       17758                 75          0
1161022   2869702        4920                 74          1
550847    1355077       34027                 74          1
1138782   2813632       44267                 74          1
595804    1465173       33055                 74          0
127782     312611       20231                 74          1
388831     949182       40198                 74          1
139742     341238       12456                 74          0
567825    1395075       13870                 74          0
139741     341238       16965                 73          0
550846    1355077       26698                 73          0
...           ...         ...                ...        ...
536430    1317906       39301                  1          0
536417    1317878        8518                  1          0
536404    1317864       40541                  1          1
536396    1317841       30814                  1          1
536393    1317833       30320                  1          1
536386    1317826       24025                  1          1
536516    1318200       35221                  1          1
536537    1318274       11577                  1          1
1206913   2980949       22481                  1          0
536543    1318325       15324                  1          1
1206914   2980996       41997                  1          0
1206917   2981036       12149                  1          0
536674    1318796       33318                  1          1
1206922   2981047       13176                  1          1
536664    1318792       36962                  1          1
536655    1318717       33704                  1          1
536646    1318662       27966                  1          0
536642    1318601       25256                  1          1
536633    1318593       19173                  1          0
536622    1318589       14540                  1          1
1206941   2981054       19660                  1          1
536613    1318579         196                  1          1
1206947   2981057       31371                  1          1
536589    1318535       24852                  1          1
1206953   2981065       13080                  1          1
536571    1318485       45603                  1          1
536569    1318378        5449                  1          1
536562    1318341        9186                  1          0
536559    1318327       16813                  1          1
0               1       49302                  1          1

[1384617 rows x 4 columns]

In [31]:
mostadded= data.loc[data['add_to_cart_order'] == 80]

In [32]:
mostadded

order_id  product_id  add_to_cart_order  reordered
567831    1395075       21641                 80          0
1138788   2813632       41771                 80          1

In [33]:
mostadded.product_id

567831     21641
1138788    41771
Name: product_id, dtype: int64

## Answer: 
### Product ID's 21641 and 41771 are added to the cart the most. 

# Q2) What are these products' names? 

In [45]:
# we need to read the csv file for the product names to merge it with the current data. 
names=pd.read_csv("products.csv")

In [36]:
names.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  
0             19  
1             13  
2              7  
3              1  
4             13

In [40]:
data = pd.merge(data, names, on='product_id', how='left')
data.head()

order_id  product_id  add_to_cart_order  reordered  \
0         1       49302                  1          1   
1         1       11109                  2          1   
2         1       10246                  3          0   
3         1       49683                  4          0   
4         1       43633                  5          1   

                                    product_name  aisle_id  department_id  
0                               Bulgarian Yogurt       120             16  
1  Organic 4% Milk Fat Whole Milk Cottage Cheese       108             16  
2                          Organic Celery Hearts        83              4  
3                                 Cucumber Kirby        83              4  
4           Lightly Smoked Sardines in Olive Oil        95             15

In [43]:
mostadded= data.loc[data['add_to_cart_order'] == 80]

In [44]:
mostadded

order_id  product_id  add_to_cart_order  reordered  \
567831    1395075       21641                 80          0   
1138788   2813632       41771                 80          1   

                              product_name  aisle_id  department_id  
567831           Mixed Nuts Lightly Salted       117             19  
1138788  Fresh European Style Baby Spinach       123              4

## Answer: The most added items are "Mixed Nuts Lightly Salted" and "Fresh European Style Baby Spinach." 